### PR TITLE
Add jsdocs to interface methods

### PR DESCRIPTION
**Pull request by jojo-stripe**
Adding docs around the methods that we surface

### CODE REVIEW

File: types/shared.d.ts [hunk 1]
```diff
@@ -116,12 +116,32 @@ export interface IStripeConnectInitParams {
 }
 
 export interface StripeConnectWrapper {
```
Comment (jorgea-stripe) [inline types/shared.d.ts:118]: This adds JSDOCS to methods, can we also add to the parameters above? I'm ok with doing this in another PR or descoping it for GA, but I do think having this is important!

Comment (jorgea-stripe) [reply types/shared.d.ts:118]: You can scope out `appearance` for now, that one we can take care of separately, since a lot of those will be renamed

**Review by jorgea-stripe — APPROVED** [review]
LGTM, suggesting documenting the parameters above too, except the ones within appearance.

File: types/shared.d.ts [hunk 2]
```diff
@@ -116,12 +116,32 @@ export interface IStripeConnectInitParams {
...
+   * @returns A promise that resolves when the user is logged out.
+   */
   logout: () => Promise<void>;
 }
 
 export interface StripeConnectInstance {
+  /**
+   * Creates a Connect element.
+   * @tagName Name of the Connect element to create.
+   * @returns An HTML Connect element
```
Comment (jorgea-stripe) [inline types/shared.d.ts:137]:
```suggestion
   * @returns An HTML component corresponding to that connect component
```

File: types/shared.d.ts [hunk 3]
```diff
@@ -116,12 +116,32 @@ export interface IStripeConnectInitParams {
...
+   * Logs the user out of Connect JS sessions
+   * @returns A promise that resolves when the user is logged out.
+   */
   logout: () => Promise<void>;
 }
 
 export interface StripeConnectInstance {
+  /**
+   * Creates a Connect element.
+   * @tagName Name of the Connect element to create.
```
Comment (jorgea-stripe) [inline types/shared.d.ts:136]:
```suggestion
   * @tagName Name of the Connect component to create.
```

File: types/shared.d.ts [hunk 4]
```diff
@@ -116,12 +116,32 @@ export interface IStripeConnectInitParams {
 }
 
 export interface StripeConnectWrapper {
+  /**
+   * Initializes a Connect JS instance that can be used to create and update Connect elements.
```
Comment (jorgea-stripe) [inline types/shared.d.ts:120]:
```suggestion
   * Initializes a Connect JS instance that can be used to create and update Connect components.
```

**Review by jorgea-stripe — APPROVED** [review]

File: types/shared.d.ts (hunk 1 above)
Comment (jojo-stripe) [reply types/shared.d.ts:118]: Will add in a followup PR

**Pull request merged**